WordPress/Classes/Stores/StatsPeriodStore.swift

```diff
@@ -397,12 +399,20 @@ private extension StatsPeriodStore {
         }
     }
 
-    func refreshPeriodOverviewData(date: Date, period: StatsPeriodUnit) {
+    func refreshPeriodOverviewData(date: Date, period: StatsPeriodUnit, forceRefresh: Bool) {
         // The call to `persistToCoreData()` might seem unintuitive here, at a first glance.
         // It's here because call to this method will usually happen after user selects a different
         // time period they're interested in. If we only relied on calls to `persistToCoreData()`
         // when user has left the screen/app, we would possibly lose on storing A LOT of data.
         persistToCoreData()
+
+        if forceRefresh {
+            // Stop existing queries and start fresh.
+            statsServiceRemote?.wordPressComRestApi.invalidateAndCancelTasks()
+            setAllAsFetchingOverview(fetching: false)
+            initializeStatsRemote()
```

Comment:
why do we need to renitialize the remote when force-refreshing?

Reply:
Hey @jklausa . `invalidateAndCancelTasks` invalidates the `SessionManager`. So the remote has to be re-init'ed for queries to run. I've added a comment explaining it.

Reply:
Ah! I misread the method as "invalidate and cancel all tasks", whereas it's meant as "invalidate the session, and cancel all tasks". gotcha now!